teps/0131-tekton-conformance-policy.md

```diff
@@ -0,0 +1,268 @@
+
+For now, only Tekton Pipelines Conformance API Specification is defined (APIs of Triggers and Results are out of scope): a ***Conformant Pipeline Service*** must be able to accept a `v1.PipelineRun`, `v1.TaskRun`, within where `v1.Task` and `v1.Pipeline` can be incorporated by reference.
+
+Tekton users are able to reuse tasks from Hub/ArtifactHub across ***conformant Tekton Pipelines Services*** or bring ***Conformant Tekton Pipelines Primitives*** from one conformant Tekton Pipelines Service to another.
+
+
+### Conformance API Specification
+
+WIP, will iterate based on https://github.com/tektoncd/pipeline/blob/main/docs/api-spec.md
+
+
+### Conformance Policy Versioning
+
+There are three kinds of changes (maybe more! feel free to propose) could happen:
+1. **GA API Version**: in the future, we may want to define Tekton V2 API.
+1. **The Conformance API**: we promote a field from `OPTIONAL` to `REQUIRED`.
+1. **The Policy Itself**: for example,  in the future, we may want to define several conformance levels that vendors can choose to conform to, or we may change the definition of the conformance.
+
+The versioning format: `<Tekton Primitive Type>-<GA API Version>.<Policy / API  Update>`
+* **pipelines.v1.0**: the initial version of **Tekton Pipelines V1 Conformance Policy**.
+* **pipelines.v2.1**: added a new required field in **Tekton Pipelines V2 Conformance APIs**.
+* **results.v2.0**: the initial version of **Tekton Results V2 Conformance Policy**.
+* **results.v2.1**: some policy change on **Tekton Results V2 Conformance Policy**.
+
+Worth noting that even if we bump up the policy version, vendors can still claim the conformance on a previous version.
+
+### Policy Update Procedure
+* **Open a PR** to propose the update.
+* **The PR** must be approved by more than half of [the project OWNERS][project owners] (i.e. 50% + 1) when it involves actual requirements changes (as opposed to typo/grammar fixing).
+* **Update Notice** ahead of time.
+  * Nice to have: conformance test kit can pop up the update notice when vendors run it.
+  * How long do we want the update notice to be ahead of is an open question.
+
+### Conformance Test Suites
+
+The general idea is we follow "[duck test][duck test]", we provide inputs (YAMLs including all required fields) and outputs (status, results etc.) to see if vendors meet the requirements (vendors will need to convert the YAMLs to the format they support). Vendors are required to open a PR containing an instruction doc for reproducing the test result, and the community reviews and approves the PR (Borrowed the idea from [Kubernetes Conformance Test][Kubernetes Conformance Test]). This is a one-time test, but the test result MUST be reproducible when the Conformance claim is valid.
```

Comment:
I don't think we should test the execution logs, because logs is very implementation-driven 🤔

For outputs, to start small, the two values we should check:
- Results
- Status

And yes we should check if DAG functions well -- by verifying certain results. wdyt?